src/rgw/rgw_op.cc

```diff
@@ -5234,6 +5234,56 @@ int RGWCopyObj::verify_permission(optional_yield y)
 					      RGW_PERM_READ)) { 
 	  return -EACCES;
 	}
+
+  // Check for tagging permissions
+  bool is_put_tagging_permission = false, is_get_tagging_permission = false, without_tagging_directive = false;
```

Comment:
Instead of multiple variables, can define Enums and set value of tagging_permission and tagging_directive vars from those enums, based on different checks here

> enum TaggingPermission {
  NO_TAGGING_PERM,
  PUT_TAGGING_PERM,
  GET_TAGGING_PERM
};

> enum TaggingDirective {
  NO_TAGGING_DIRECTIVE,
  COPY_TAGGING_DIRECTIVE,
  REPLACE_TAGGING_DIRECTIVE
};

Reply:
Done changes as suggested.

Reply:
all new enum decls should be enum class

Reply:
> all new enum decls should be enum class
@shraddhaghatol can you please address this